feat: handle postman import with same folder names (#955)

Jofresh opened the pull request:
# Description

Refers to https://github.com/usebruno/bruno/issues/955 

If you have multiple folders with the same names when you import your Postman Collection, only the first one will be imported and all of the others will be ignored. I think there are 3 solutions 
1. Merge all the requests into one folder
2. Keep the folders and add a number to the end
3. Simply notify the user that all the folders could not be imported

For now I chose the 2nd solution, see capture : 

https://github.com/usebruno/bruno/assets/46541094/91792067-9669-410f-a12f-94d9ca8585ea

Let me know if it's the desired behaviour.

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

helloanoop commented:
Thanks for taking care of this @Jofresh !

helloanoop commented:
@Jofresh I made a small change https://github.com/usebruno/bruno/commit/553d1c062e0a71b947a30fee1566538689f671cf

I moved the `folderMap` inside the item iteration, otherwise it will consider folder names as duplicate even when they are different parent folders.

Hope this is correct?

Jofresh commented:
Hello @helloanoop ! Thank you for merging my contribution 😄
 
You're totally right, if they are different parent folders, this will be duplicated. Unfortunately, I tested your small change on your commit 553d1c0 and when I import a Postman collection, the initial problem appears (only one folder appears). For now, I don't see how I can fix this, but I'm working on it. 

Let me know if you have a solution or if you're already working on it 😃

helloanoop commented:
I am not @Jofresh 

You are in a better position to fix this as you have a sample to test.
You can raise a PR once you have the fix.

Jofresh commented:
Ok I'm working on it, I will raise a PR once it's fixed

Jofresh commented:
> @Jofresh I made a small change [553d1c0](https://github.com/usebruno/bruno/commit/553d1c062e0a71b947a30fee1566538689f671cf)
> 
> I moved the `folderMap` inside the item iteration, otherwise it will consider folder names as duplicate even when they are different parent folders.
> 
> Hope this is correct?

In the end, I figured out that file names were not considered as duplicates even if they are different parent folders, in other words a folder "A" included in a folder "A" will not be renamed "A_1" as they are not from the same root. In fact, since our function is called recursively when a folder is involved, the "folderMap" object is reassigned to an empty object and so a folder "A" in a folder "A" will not be considered a duplicate (and renamed A_1).

Here is a preview with nested folders : 

<img width="196" alt="image" src="https://github.com/usebruno/bruno/assets/46541094/edbe6fb6-ff18-46b5-84c4-c511f9fa905b">

So the folderMap needs to stay at the top of the for-each loop, as in my initial commit 7980b726f67198179694998e2d98cdf921e4677d.

**Sorry for the confusion and for wasting some of your time 😅.**

_I was able to test with different Postman collections and the result seems correct. If that's not the result you want and you want to test to be sure, here are the samples I used :_
- _The first one with 5 folders with the same name : https://pastebin.com/dTXwUiPt_
- _The second one with nested folders named the same : https://pastebin.com/uimXEYDW_

Thank you again for considering my contribution 🙂

helloanoop commented:
I reverted my commit.

> Sorry for the confusion and for wasting some of your time 😅.

Well, I think I caused the problem here and wasted your time 😅

Thanks for all your efforts @Jofresh 👊💛
